omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/ModuleActionResource1_8.java

```diff
@@ -0,0 +1,225 @@
+	}
+	
+	/**
+	 * Overriding create directly, because ModuleFactory requires ServletContext to execute any
+	 * action
+	 */
+	@Override
+	public Object create(SimpleObject post, RequestContext context) throws ResponseException {
+		moduleFactoryWrapper.checkPrivilege();
+		
+		ModuleAction action = newDelegate();
+		setConvertedProperties(action, post, getCreatableProperties(), true);
+		
+		Collection<Module> modules;
+		if (action.isAllModules() != null && action.isAllModules()) {
+			modules = moduleFactoryWrapper.getLoadedModules();
+			action.setModules(new ArrayList<Module>(modules));
+		} else {
+			modules = action.getModules();
+		}
+		
+		ServletContext servletContext = getServletContext(context);
+		
+		if (modules == null || modules.isEmpty()) {
+			throw new IllegalRequestException("Cannot execute action " + action.getAction() + " on empty set of modules.");
+		} else {
+			//rest module cannot be stopped/unloaded during handling request
+			Module restModule = moduleFactoryWrapper.getModuleById(RestConstants.MODULE_ID);
+			modules.remove(restModule);
+			switch (action.getAction()) {
+				case START:
+					startModules(modules, servletContext);
+					break;
+				case STOP:
+					stopModules(modules, servletContext, true);
+					break;
+				case RESTART:
+					restartModules(modules, servletContext);
+					break;
+				case UNLOAD:
+					unloadModules(modules, servletContext);
+					break;
+			}
+		}
+		
+		return ConversionUtil.convertToRepresentation(action, Representation.DEFAULT);
+	}
+	
+	private void restartModules(Collection<Module> modules, ServletContext servletContext) {
+		stopModules(modules, servletContext, false);
+		startModules(modules, servletContext);
+	}
+	
+	private void unloadModules(Collection<Module> modules, ServletContext servletContext) {
+		boolean needsRefresh = false;
+		for (Module module : modules) {
+			if (moduleFactoryWrapper.isModuleStarted(module)) {
+				moduleFactoryWrapper.stopModuleSkipRefresh(module, servletContext);
+				needsRefresh = true;
+			}
+			moduleFactoryWrapper.unloadModule(module);
+		}
+		
+		if (needsRefresh) {
+			moduleFactoryWrapper.refreshWebApplicationContext(servletContext);
+		}
+	}
+	
+	/**
+	 * @param modules modules to stop
+	 * @param servletContext ServletContext is required by WebModuleUtil to perform operation
+	 */
+	private void stopModules(Collection<Module> modules, ServletContext servletContext, boolean refreshContext) {
+		for (Module module : modules) {
+			if (moduleFactoryWrapper.isModuleStarted(module)) {
+				moduleFactoryWrapper.stopModuleSkipRefresh(module, servletContext);
+			}
+		}
+		if (refreshContext) {
+			moduleFactoryWrapper.refreshWebApplicationContext(servletContext);
+		}
+	}
+	
+	/**
+	 * @param modules modules to start
+	 * @param servletContext ServletContext is required by WebModuleUtil to perform operation
+	 */
+	private void startModules(Collection<Module> modules, ServletContext servletContext) {
```

Comment:
@PawelGutkowski i have just installed the modules you mentioned above, stopped metadatasharing, and then stopped metadatamapping. Then i sent a post request with the body below: 
{   "allModules": "true",   "action": "start" }
It has worked seamlessly. Is there anything am missing?